stl/inc/tuple

```diff
@@ -678,6 +778,58 @@ _NODISCARD constexpr common_comparison_category_t<_Synth_three_way_result<_Types
     static_assert(sizeof...(_Types1) == sizeof...(_Types2), "cannot compare tuples of different sizes");
     return _Left._Three_way_compare(_Right);
 }
+
+#if _HAS_CXX23
+template <class _TTuple, class _UTuple, class _Indices = make_index_sequence<tuple_size_v<_TTuple>>>
+struct _Can_use_get_for_all_indices;
+
+template <class _TTuple, class _UTuple, size_t... _Indices>
+struct _Can_use_get_for_all_indices<_TTuple, _UTuple, index_sequence<_Indices...>>  : bool_constant<
+    (requires(const _TTuple& _Left, const _UTuple& _Right) {
+    { _STD get<_Indices>(_Left) == _STD get<_Indices>(_Right) } -> _Boolean_testable; } && ...)> {};
+
+template <class _TTuple, class _UTuple, size_t... _Indices>
+_NODISCARD constexpr bool _Equality_compare_tuples(
+    const _TTuple& _Left, const _UTuple& _Right, index_sequence<_Indices...>) {
+    return ((_STD get<_Indices>(_Left) == _STD get<_Indices>(_Right)) && ...);
+}
+
+_EXPORT_STD template <class... _TTypes, _Tuple_like _UTuple>
+_NODISCARD constexpr bool operator==(const tuple<_TTypes...>& _Left, const _UTuple& _Right) {

```

Comment:
This `operator==` needs to be a hidden friend, ditto for the new `operator<=>` ([[tuple.rel]/5.2](https://eel.is/c++draft/tuple.rel#5.2), [[tuple.rel]/8](https://eel.is/c++draft/tuple.rel#8)).

However, the current standard woring is a bit weird - it's unspecified in which class body should these operators be defined. And if the intent is defining them in `tuple`, then `tuple<TTypes...>` is redundant (as injected-class-name `tuple` is sufficient).

I suppose there should be an LWG issue for this. CC @cor3ntin.

Reply:
Done!

Reply:
@frederick-vs-ja Yes, I think I agree that ideally the second overload should be declared as a hidden friend in the class declaration of the wording synopsis. I'll talk to LWG